route/time_dependent.go

```diff
@@ -0,0 +1,118 @@
+package route
+
+import (
+	"errors"
+	"sort"
+
+	"github.com/nextmv-io/sdk/model"
+)
+
+// ByIndexAndTime holds a Measure and an EndTime (exclusive) up until this
+// measure is to be used as a Unix timestamp. ByIndexAndTime is to be used with
+// NewTimeDependentMeasure which a slice of ByIndexAndTime.
+type ByIndexAndTime struct {
+	Measure ByIndex
+	EndTime int
+}
+
+type client struct {
+	measures        []ByIndexAndTime
+	fallbackMeasure ByIndexAndTime
+	cache           map[int]ByIndexAndTime
```

Comment:
Instead of mapping to a reference to a `ByIndexAndTime` you could also map to the measure index (`map[int]int`). This is a bit more memory efficient I would expect.

Reply:
I know, we discussed it. but when I implemented it it felt more natural to use a map because the index is not starting from 0 and needs a shift from the start time. I can do that but it feels weird.

Reply:
I am not suggesting to use a slice, but a map from int to int instead of int to ByIndexAndTime.